File: Pythonleetcode/Leetcode48.py
def rotate(matrix):
    n = len(matrix)
    for i in range(0, n - 1):
        for j in range(i, n - 1):
            if i <= ((n - 1) / 2) and j <= n - i - 2:
                matrix[i][j], matrix[j][n - 1 - i], matrix[n - 1 - i][n - 1 - j], matrix[n - 1 - j][i] = \
                matrix[n - 1 - j][i], matrix[i][j], matrix[j][n - 1 - i], matrix[n - 1 - i][n - 1 - j]

    return matrix





matrix=[[5,1,9,11],[2,4,8,10],[13,3,6,7],[15,14,12,16]]
print(matrix[0][0])
print(rotate(matrix))

File: CSharpPractice/LeetcodePractice/Leetcode48.cs
public class Solution48 {
    public void Rotate(int[][] matrix) 
    {
        int n = matrix.Length;
        int topLeft = 0;
        int topRight = 0;
        int bottomRight = 0;
        int bottomLeft = 0;
        for(int i=0;i<=n-2;i++)
        {
            for(int j=i;j<=n-2-i;j++)
            {
                topLeft = matrix[i][j];
                topRight = matrix[j][n-1-i];
                bottomRight  = matrix[n-1-i][n-1-j];
                bottomLeft = matrix[n-1-j][i];

                matrix[i][j] = bottomLeft ;
                matrix[j][n-1-i] = topLeft;
                matrix[n-1-i][n-1-j] = topRight;
                matrix[n-1-j][i]=bottomRight;
            }
        }
    }
}